plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/model/PreconditionError.java

```diff
@@ -0,0 +1,118 @@
+package io.jenkins.tools.pluginmodernizer.core.model;
+
+import java.util.function.BiFunction;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import org.w3c.dom.Document;
+
+/**
+ * Enum to represent the precondition errors preventing any modernization process
+ * Generally, these are the errors that need to be fixed before applying any modernization (very old plugin)
+ * We can provide in future version a way to fix these errors automatically (without OpenRewrite) by adding a fix function
+ * on this enum
+ */
+public enum PreconditionError {
+
+    /**
+     * No pom file found
+     */
+    NO_POM(
+            (document, xpath) -> {
+                return document == null;
+            },
+            "No pom file found"),
+
+    /**
+     * If the plugin has HTTP repositories preventing modernization
+     */
+    MAVEN_REPOSITORIES_HTTP(
+            (document, xpath) -> {
+                if (document == null) {
+                    return false;
+                }
+                try {
+                    Double nonHttpsRepositories = (Double) xpath.evaluate(
+                            "count(//*[local-name()='project']/*[local-name()='repositories']/*[local-name()='repository']/*[local-name()='url' and not(starts-with(., 'https'))])",
+                            document,
+                            XPathConstants.NUMBER);
+                    return nonHttpsRepositories != null && !nonHttpsRepositories.equals(0.0);
+                } catch (Exception e) {
+                    return false;
+                }
+            },
+            "Found non-https repository URL in pom file preventing maven older than 3.8.1"),
+
+    /**
+     * If the plugin has an older Java version preventing modernization
+     */
+    OLDER_JAVA_VERSION(
+            (document, xpath) -> {
+                if (document == null) {
+                    return false;
+                }
+                try {
+                    String javaVersion = (String) xpath.evaluate(
+                            "//*[local-name()='project']/*[local-name()='properties']/*[local-name()='java.level']",
+                            document,
+                            XPathConstants.STRING);
+                    return javaVersion != null && (javaVersion.equals("7") || javaVersion.equals("6"));
```

Comment:
There are some plugins that use Java 5 too.
Eg: https://github.com/jenkinsci/groovy-label-assignment-plugin